src/NuGet.Clients/VisualStudio/Extensibility/IVsPackageProjectMetadata.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Runtime.InteropServices;
+
+namespace NuGet.VisualStudio
+{
+    /// <summary>
+    /// contain information about project where we're executing nuget actions
+    /// </summary>
+    [ComImport]
+    [Guid("7ce0caae-1b99-4cd1-80bf-348972975bcf")]
+    public interface IVsPackageProjectMetadata
+    {
+        /// <summary>
+        /// guid for batch events of the project
+        /// </summary>
+        string Guid { get; }
+
+        /// <summary>
+        /// name of the project
+        /// </summary>
+        string Name { get; }
+
+        /// <summary>
+        /// full path of the project
+        /// </summary>
+        string FullPath { get; }
+
+        /// <summary>
+        /// target framework of the project
+        /// </summary>
+        string TargetFramework { get; }
```

Comment:
This interface should be as minimal as possible. The extra information is nice but populating all of this information for every operation, including when there is no one listening can be a problem.

Remove target framework, I don't think this is needed and it doesn't make sense for projects with multiple target frameworks.

FullPath should always be available, but if you can drop it I would. This event will fire during template installs where the file may not even be on disk yet, or may not have a path. Discovering the project path from DTE in scenarios like that can be problematic.